cmd/scollector/collectors/rabbitmq.go

```diff
@@ -0,0 +1,437 @@
+		}
+
+	}
+	return md, nil
+}
+
+func rabbitmqMessageStats(p string, ts opentsdb.TagSet, ms rmqMessageStats) opentsdb.MultiDataPoint {
+	var md opentsdb.MultiDataPoint
+	Add(&md, p+"message_stats", ms.Ack, ts.Copy().Merge(opentsdb.TagSet{"method": "ack"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsAck)
+	Add(&md, p+"message_stats", ms.Confirm, ts.Copy().Merge(opentsdb.TagSet{"method": "confirm"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsConfirm)
+	Add(&md, p+"message_stats", ms.Deliver, ts.Copy().Merge(opentsdb.TagSet{"method": "deliver"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsDeliver)
+	Add(&md, p+"message_stats", ms.DeliverGet, ts.Copy().Merge(opentsdb.TagSet{"method": "deliver_get"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsDeliverGet)
+	Add(&md, p+"message_stats", ms.DeliverNoAck, ts.Copy().Merge(opentsdb.TagSet{"method": "deliver_noack"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsDeliverNoAck)
+	Add(&md, p+"message_stats", ms.Get, ts.Copy().Merge(opentsdb.TagSet{"method": "get"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsGet)
+	Add(&md, p+"message_stats", ms.GetNoAck, ts.Copy().Merge(opentsdb.TagSet{"method": "get_noack"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsGetNoack)
+	Add(&md, p+"message_stats", ms.Publish, ts.Copy().Merge(opentsdb.TagSet{"method": "publish"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsPublish)
+	Add(&md, p+"message_stats", ms.PublishIn, ts.Copy().Merge(opentsdb.TagSet{"method": "publish_in"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsPublishIn)
+	Add(&md, p+"message_stats", ms.PublishOut, ts.Copy().Merge(opentsdb.TagSet{"method": "publish_out"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsPublishOut)
+	Add(&md, p+"message_stats", ms.Redeliver, ts.Copy().Merge(opentsdb.TagSet{"method": "redeliver"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsRedeliver)
+	Add(&md, p+"message_stats", ms.Return, ts.Copy().Merge(opentsdb.TagSet{"method": "return"}),
+		metadata.Counter, metadata.Message, DescRmqMessageStatsReturn)
+	return md
+}
+
+func rabbitmqBackingQueueStatus(p string, ts opentsdb.TagSet, bqs rmqBackingQueueStatus) opentsdb.MultiDataPoint {
+	var md opentsdb.MultiDataPoint
+	Add(&md, p+"avg_rate", bqs.AvgAckEgressRate, ts.Copy().Merge(opentsdb.TagSet{"method": "ack", "direction": "out"}),
+		metadata.Rate, metadata.Message, DescRmqBackingQueueStatusAvgAckEgressRate)
+	Add(&md, p+"avg_rate", bqs.AvgAckEgressRate, ts.Copy().Merge(opentsdb.TagSet{"method": "ack", "direction": "in"}),
+		metadata.Rate, metadata.Message, DescRmqBackingQueueStatusAvgAckIngressRate)
+	Add(&md, p+"avg_rate", bqs.AvgAckEgressRate, ts.Copy().Merge(opentsdb.TagSet{"direction": "out"}),
+		metadata.Rate, metadata.Message, DescRmqBackingQueueStatusAvgEgressRate)
+	Add(&md, p+"avg_rate", bqs.AvgAckEgressRate, ts.Copy().Merge(opentsdb.TagSet{"direction": "in"}),
```

Comment:
You are adding the same two data points twice, with different tagsets (once with direction and method, once with only direction). This scheme seems like it may cause confusing data conditions. Why Not have just the first two Adds and the last one?